src/ngraph_builder.cc

```diff
@@ -446,6 +447,37 @@ static Status TranslateAddNOp(
   return Status::OK();
 }
 
+static Status TranslateArgMaxOp(
+    const Node* op, const std::vector<const Tensor*>& static_input_map,
+    Builder::OpMap& ng_op_map) {
+  shared_ptr<ng::Node> ng_input, ng_dim;
+
+  TF_RETURN_IF_ERROR(GetInputNodes(ng_op_map, op, &ng_input, &ng_dim));
+
+  std::vector<int64> tf_dim;
+  TF_RETURN_IF_ERROR(GetStaticInputVector(op, 1, static_input_map, &tf_dim));
+
+  ng::Shape input_shape = ng_input->get_shape();
+  size_t input_rank = ng_input->get_shape().size();
+
+  // If input dimension is negative, make it positive
+  if (tf_dim[0] < 0) {
+    tf_dim[0] = input_rank + tf_dim[0];
+  }
+  size_t input_dims = tf_dim[0];
+
+  DataType dtype = DT_INT64;
+  TF_RETURN_IF_ERROR(GetNodeAttr(op->attrs(), "output_type", &dtype));
```

Comment:
We need to read the datatype from the op. It could be DT_INT32 too. 

Something like:
```
DataType dtype;
TF_RETURN_IF_ERROR(GetNodeAttr(op->attrs(), "dtype", &dtype));
```

Reply:
This is the expected data type of the output set by the user as an attribute. Not sure why we need to read datatype of the op.

Reply:
Right. By 'datatype of the op', I meant 'datatype of the output set by the user'

Reply:
This is the output data type attr used in TF '''DataType tensorflow::ops::ArgMax::Attrs::output_type_''' and the function '''TF_MUST_USE_RESULT Attrs tensorflow::ops::ArgMax::Attrs::OutputType(
  DataType x
)'''

Reply:
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/ops/math_ops.cc#L896

DT_INT64 is the default if nothing is specified. But it could be DT_INT32, hence we need to read that.

So this looks good.